include/seastar/core/future.hh

```diff
@@ -326,6 +337,10 @@ public:
 template <typename T> struct uninitialized_wrapper_base<T, true> : private T {
     using tuple_type = future_tuple_type_t<T>;
     uninitialized_wrapper_base() noexcept = default;
+    template<typename U>
+    void uninitialized_set(implicit_tag, U&& v) {
+        new (this) T(id<T>(std::forward<U>(v)));
+    }
```

Comment:
AFAIK placement new won't perform an excessive copy or move from an rvalue of the same type returned by `id` function.
This will need some performance testing.